pkg/service/service.go

```diff
@@ -684,7 +684,12 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 		}
 	}
 
-	// Update lbmaps (BPF service maps)
+	// If there are no frontend IP (i.e. headless service), we should return immediately
+	// and avoid plumbing into BPF lb maps
+	if params.Type == lb.SVCTypeNone {
+		return false, lb.ID(0), nil
+	}
```

Comment:
I'm a bit concerned that the delete path might not work properly; I would expect that the deletion of the headless service from the `lbmap` \[1\] fails given that it had not been inserted in the first place \[2\]. Is it that the case?

\[1\]: https://github.com/cilium/cilium/blob/4e794b7304da0e20851807f4836c0f1c2e9ee728/pkg/service/service.go#L1579-L1582
\[2\]: https://github.com/cilium/cilium/blob/4e794b7304da0e20851807f4836c0f1c2e9ee728/pkg/service/service.go#L689-L698

Reply:
you are right, I overlooked the warning log if the lbmap file is not found.